presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -239,7 +240,9 @@ private synchronized void createGroupIfNecessary(SelectionContext<C> context, Ex
                 createGroupIfNecessary(new SelectionContext<>(id.getParent().get(), context.getContext()), executor);
                 InternalResourceGroup parent = groups.get(id.getParent().get());
                 requireNonNull(parent, "parent is null");
-                group = parent.getOrCreateSubGroup(id.getLastSegment());
+                // parent segments size equals to subgroup segment index
+                int subGroupSegmentIndex = parent.getId().getSegments().size();
+                group = parent.getOrCreateSubGroup(id.getLastSegment(), (context.getFirstDynamicSegmentPosition().equals(OptionalInt.of(subGroupSegmentIndex))));
```

Comment:
i think this logic works to identify the given resource group is static or not. But a much cleaner way would be, if we mark each `segment` of `ResourceGroupId` being static or not when we create them via `group.expandTemplate` call, then we can easily put the flag here. And no need to rely on the context. 

i.e. this line will become something like this
group = parent.getOrCreateSubGroup(id.getLastSegment().getValue(), id.getLastSegment().isStatic());

And i think to make this happen, we may need to change the `segments` in `ResourceGroupId` class from string to a class Segment which looks like this:
class Segment {
    String value;
    String isStatic;
}

Reply:
We had a offline chat with @swapsmagic. We concluded that introducing a segment class would require a major code change. We are going to keep it as is.